tests/zfs-tests/tests/functional/mmp/mmp_hostid.ksh

```diff
@@ -42,6 +42,9 @@ function cleanup
 	default_cleanup_noexit
 	log_must rm $MMP_DIR/file.{0,1,2,3,4,5}
 	log_must rmdir $MMP_DIR
+	if [[ -L $HOSTID_FILE ]]; then
+		rm -f $HOSTID_FILE
+	fi
```

Comment:
Is the concern that this should only be removed when it's a link?  If so, that seems reasonable, but you'll want to move `mmp_clear_hostid` call in to this conditional since it removes this file.

Reply:
well, i have found `ln -s` to file in dataset, but link not removed - it is why this check available there - if we exit from test by mistake, link can be present

Reply:
Ahh I see.  Possibly the `log_must` calls on the two previous lines were preventing the code from reaching here?  We've seen this before, a more robust fix might be to remove then in this cleanup code so we always cleanup everything we can.  Then this _shouldn't_ be needed.